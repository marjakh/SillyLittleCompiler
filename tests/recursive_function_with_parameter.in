function foo(i) {
  write(100);
  write(i);
  i = i - 1;
  if (i > 0) {
    foo(i);
  }
  write(200);
  write(i);
}

foo(5);

function bar(i) {
  write(300);
  write(i);
  if (i > 0) {
    bar(i - 1);
  }
  write(400);
  write(i);
}

bar(5);