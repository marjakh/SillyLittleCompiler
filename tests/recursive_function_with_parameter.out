100
5
100
4
100
3
100
2
100
1
200
0
200
1
200
2
200
3
200
4
300
5
300
4
300
3
300
2
300
1
300
0
400
0
400
1
400
2
400
3
400
4
400
5
